### PR TITLE
VZ-2189: Fix ES logging when app has multiple workload types

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

This is a short/medium-term fix for ES logging when an application is compromised of multiple workload types. Previously we created a single shared fluentd configmap for all pods regardless of workload type. This change results in a separate configmap per workload type.

I tested by running multiple examples and looking in Kibana to make sure I saw logs for all pods. For Bob's Books, I confirmed that both WebLogic and Coherence logs were sent. I also added another test to the Bob's Books test suite to look for Coherence log records in addition to WebLogic log records.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/loggingscope/fluentd.go [hunk 1]
```diff
@@ -161,7 +162,7 @@ func (f *Fluentd) ensureFluentdVolumeMountExists(fluentdPod *FluentdPod) {
 // to do. If it doesn't exist, create it.
 func (f *Fluentd) ensureFluentdConfigMapExists(namespace string) error {
 	// check if configmap exists
-	configMapExists, err := resourceExists(f.Context, f, configMapAPIVersion, configMapKind, configMapName, namespace)
+	configMapExists, err := resourceExists(f.Context, f, configMapAPIVersion, configMapKind, configMapName+"-"+f.WorkloadType, namespace)
```
Comment (kminder) [inline application-operator/controllers/loggingscope/fluentd.go:165]: I'm surprised go fmt doesn't have an opinion about space vs no-space around string operators.  You have two different styles in this change.

Comment (kminder) [reply application-operator/controllers/loggingscope/fluentd.go:165]: @fredtibbitts confirmed that this spacing is what go fmt requires.  Resolved.

**Review by kminder — APPROVED** [review]
Approved.

File: application-operator/controllers/loggingscope/fluentd.go (hunk 1 above)
Comment (fredtibbitts) [reply application-operator/controllers/loggingscope/fluentd.go:165]: That is `gofmt`ed. It does use inconsistent rules for string concatenation (function call vs struct field value), but apparently that's on purpose. 🤷

**Pull request merged**